### PR TITLE
Put a 2 layer cache in the gateway.

### PR DESCRIPTION
**Pull request by ianopolous**
This reduces typical response time from 100-700ms down to 7-50ms

### CODE REVIEW

File: src/peergos/server/net/GatewayHandler.java [hunk 1]
```diff
@@ -16,15 +16,46 @@
 public class GatewayHandler implements HttpHandler {
 	private static final Logger LOG = Logging.LOG();
 	private static final boolean LOGGING = true;
+	private static final int MAX_ASSET_SIZE_CACHE = 200*1024;
 
     private final String domainSuffix;
     private final NetworkAccess network;
     private final Crypto crypto;
+    private final LRUCache<String, FileWrapper> webRootCache;
```
Comment (kevodwyer) [inline src/peergos/server/net/GatewayHandler.java:24]: parameterise?

File: src/peergos/server/net/GatewayHandler.java [hunk 2]
```diff
@@ -16,15 +16,46 @@
...
+    private final LRUCache<String, Asset> assetCache;
 
     public GatewayHandler(String domainSuffix, Crypto crypto, NetworkAccess network) {
         this.domainSuffix = domainSuffix;
         this.crypto = crypto;
         this.network = network;
+        this.webRootCache = new LRUCache<>(1000);
```
Comment (kevodwyer) [inline src/peergos/server/net/GatewayHandler.java:31]: paramaterise size?

Comment (ianopolous) [reply src/peergos/server/net/GatewayHandler.java:31]: It's not random, it's chosen based on the maximum amount of ram I want the cache to use.

Comment (ianopolous) [reply src/peergos/server/net/GatewayHandler.java:31]: And the number of users we have

**Review by kevodwyer — APPROVED** [review]

**Pull request merged**